sdk/python/feast/client.py

```diff
@@ -977,197 +601,126 @@ def ingest(
-            force_refresh: Setting this flag to true will force a recalculation
-                of featureStatistics and overwrite results currently in the cache, if any.
-            project: Manual override for default project.
+        batch_source_type = SourceType(feature_table.batch_source.type).name
 
-        Returns:
-           Returns a tensorflow DatasetFeatureStatisticsList containing TFDV featureStatistics.
-        """
+        try:
+            if batch_source_type == "BATCH_FILE":
+                from urllib.parse import urlparse
 
-        if ingestion_ids is not None and (
-            start_date is not None or end_date is not None
-        ):
-            raise ValueError(
-                "Only one of dataset_id or [start_date, end_date] can be provided."
-            )
+                file_url = feature_table.batch_source.file_options.file_url[:-1]
+                uri = urlparse(file_url)
+                staging_client = get_staging_client(uri.scheme)
 
-        if project != "" and "/" not in feature_set_id:
-            feature_set_id = f"{project}/{feature_set_id}"
+                file_name = dest_path.split("/")[-1]
+                date_today = datetime.datetime.today().strftime("%Y-%m-%d")
 
-        request = GetFeatureStatisticsRequest(
-            feature_set_id=feature_set_id,
-            features=features,
-            store=store,
-            force_refresh=force_refresh,
-        )
-        if ingestion_ids is not None:
-            request.ingestion_ids.extend(ingestion_ids)
-        else:
-            if start_date is not None:
-                request.start_date.CopyFrom(
-                    Timestamp(seconds=int(start_date.timestamp()))
+                staging_client.upload_file(
+                    dest_path,
+                    uri.hostname,
+                    str(uri.path).strip("/") + "/" + f"date={date_today}/" + file_name,
                 )
-            if end_date is not None:
-                request.end_date.CopyFrom(Timestamp(seconds=int(end_date.timestamp())))
+            if batch_source_type == "BATCH_BIGQUERY":
+                from google.cloud import bigquery
+
+                bq_table_ref = feature_table.batch_source.bigquery_options.table_ref
+                gcp_project, dataset_table = bq_table_ref.split(":")
+                dataset, table = dataset_table.split(".")
+
+                client = bigquery.Client(project=gcp_project)
+
+                table_ref = client.dataset(dataset).table(table)
+                job_config = bigquery.LoadJobConfig()
+                job_config.source_format = bigquery.SourceFormat.PARQUET
+
+                # Check for date partitioning column in FeatureTable spec
+                if feature_table.batch_source.date_partition_column:
+                    time_partitioning_obj = bigquery.table.TimePartitioning(
+                        field=feature_table.batch_source.date_partition_column
```

Comment:
`date_partition_column` most probably would be specified only for files. Since BQ is smart enough to partition on timestamp column

Reply:
so we can use eventTimestamp here instead

Reply:
Updated.